llama-index-integrations/storage/kvstore/llama-index-storage-kvstore-mongodb/llama_index/storage/kvstore/mongodb/base.py

```diff
@@ -115,6 +120,47 @@ def _check_async_client(self) -> None:
         if self._adb is None:
             raise ValueError("MongoDBKVStore was not initialized with an async client")
 
+    def _process_index_struct(
```

Comment:
Wouldn't this logic be specific to the MongoDBIndexStore class?

Reply:
Feels kind of weird to have this in the lower-level kv store

Reply:
I guess it would be specific to any index store that uses a MongoDBKVStore, which might not be restricted to the MongoDBIndexStore class -- for instance in test_kvstore.py I created a general KVIndexStore that uses a MongoDBKVStore. Do you think it would make more sense to move the logic to the KVIndexStore class?